include/psa/crypto_config.h

```diff
@@ -1078,12 +1018,26 @@
  *
  * Module:  library/psa_crypto.c
  *
- * Requires: either MBEDTLS_CTR_DRBG_C and MBEDTLS_ENTROPY_C,
- *           or MBEDTLS_HMAC_DRBG_C and MBEDTLS_ENTROPY_C,
- *           or MBEDTLS_PSA_CRYPTO_EXTERNAL_RNG.
+ * Requires: one of the following:
+ *           - MBEDTLS_PSA_CRYPTO_EXTERNAL_RNG
+ *           - MBEDTLS_CTR_DRBG_C
+ *           - MBEDTLS_HMAC_DRBG_C
+ *
+ *           If MBEDTLS_CTR_DRBG_C or MBEDTLS_HMAC_DRBG_C is used as the PSA
+ *           random generator, then either PSA_WANT_ALG_SHA_256 or
+ *           PSA_WANT_ALG_SHA_512 must be enabled for the entropy module.
+ *
  * Auto-enables: MBEDTLS_CIPHER_C if any unauthenticated (ie, non-AEAD) cipher
  *               is enabled in PSA (unless it's fully accelerated, see
  *               docs/driver-only-builds.md about that).
+ *
+ * \note The PSA crypto subsystem prioritizes DRBG mechanisms as follows:
+ *       - #MBEDTLS_PSA_CRYPTO_EXTERNAL_RNG, if enabled
+ *       - CTR_DRBG (AES), seeded by the entropy module, if
+ *         #MBEDTLS_CTR_DRBG_C is enabled
+ *       - HMAC_DRBG, seeded by the entropy module
```

Comment:
This seems to suggest that if `MBEDTLS_PSA_CRYPTO_EXTERNAL_RNG` is disabled, then `MBEDTLS_HMAC_DRBG_C` will be auto-enabled. Is that what we are trying to say? If it is, then I am not sure where/how we do it?

Reply:
That's not how I read it (but I'm biased: this is very close to [what I'd written in 3.x](https://github.com/Mbed-TLS/mbedtls/blob/mbedtls-3.6.4/include/mbedtls/mbedtls_config.h#L4163)). We don't auto-enable any of `MBEDTLS_PSA_CRYPTO_EXTERNAL_RNG`, `MBEDTLS_CTR_DRBG_C` or `MBEDTLS_HMAC_DRBG_C` in the config adjustment code. We pick the first one that's enabled in the PSA RNG code.

Reply:
I mean, one bulletpoint above, we feel the need to specify that we use CTR_DRBG only if `MBEDTLS_CTR_DRBG_C` is enabled. The last bulletpoint in comparison feels unconditional.

Reply:
Ah, I see what you mean. The HMAC_DRBG bullet point in both versions should reference `MBEDTLS_HMAC_DRBG_C`.

Reply:
I've fixed that.